tests/unit/acquisition/test_function.py

```diff
@@ -967,3 +975,228 @@ def test_lipschitz_penalizers_raises_for_invalid_pending_points_shape(
+        gibbon(model, samples)
+
+
+@pytest.mark.parametrize("pending_points", [tf.constant([0.0]), tf.constant([[[0.0], [1.0]]])])
+def test_gibbon_builder_raises_for_invalid_pending_points_shape(
+    pending_points,
+) -> None:
+    data = Dataset(tf.zeros([3, 2], dtype=tf.float64), tf.ones([3, 2], dtype=tf.float64))
+    space = Box([0, 0], [1, 1])
+    builder = GIBBON(search_space=space)
+    builder.prepare_acquisition_function(
+        data, QuadraticMeanAndRBFKernel(), None
+    )  # first initialize
+    with pytest.raises(TF_DEBUGGING_ERROR_TYPES):
+        builder.prepare_acquisition_function(data, QuadraticMeanAndRBFKernel(), pending_points)
+
+
+def test_gibbon_raises_when_called_before_initialization() -> None:
+    data = Dataset(tf.zeros([3, 2], dtype=tf.float64), tf.ones([3, 2], dtype=tf.float64))
+    search_space = Box([0, 0], [1, 1])
+    pending_points = tf.zeros([1, 2])
+    with pytest.raises(ValueError):
+        GIBBON(search_space).prepare_acquisition_function(
+            data, QuadraticMeanAndRBFKernel(), pending_points
+        )
+
+
+@pytest.mark.parametrize("at", [tf.constant([[0.0], [1.0]]), tf.constant([[[0.0], [1.0]]])])
+def test_gibbon_raises_for_invalid_batch_size(at: TensorType) -> None:
+    model = QuadraticMeanAndRBFKernel()
+    gibbon_acq = gibbon(model, tf.constant([[1.0], [2.0]]))
+
+    with pytest.raises(TF_DEBUGGING_ERROR_TYPES):
+        gibbon_acq(at)
+
+
+def test_gibbon_raises_for_model_without_homoscedastic_likelihood() -> None:
+    class dummy_model_without_likelihood(ProbabilisticModel):
+        def predict(self, query_points: TensorType) -> tuple[None, None]:
+            return None, None
+
+        def predict_joint(self, query_points: TensorType) -> tuple[None, None]:
+            return None, None
+
+        def sample(self, query_points: TensorType, num_samples: int) -> None:
+            return None
+
+        def covariance_between_points(
+            self, query_points_1: TensorType, query_points_2: TensorType
+        ) -> None:
+            return None
+
+    with pytest.raises(ValueError):
+        model_without_likelihood = dummy_model_without_likelihood()
+        gibbon(model_without_likelihood, tf.constant([1.0]))
+
+
+def test_gibbon_raises_for_model_without_covariance_between_points_method() -> None:
+    class dummy_model_without_covariance_between_points(ProbabilisticModel):
+        def predict(self, query_points: TensorType) -> tuple[None, None]:
+            return None, None
+
+        def predict_joint(self, query_points: TensorType) -> tuple[None, None]:
+            return None, None
+
+        def sample(self, query_points: TensorType, num_samples: int) -> None:
+            return None
+
+        def get_observation_noise(self) -> None:
+            return None
+
+    with pytest.raises(ValueError):
+        model_without_likelihood = dummy_model_without_covariance_between_points()
+        gibbon(model_without_likelihood, tf.constant([1.0]))
```

Comment:
I'm surprised this test works (I thought it would raise an uncaught AttributeError instead as I commented earlier) but apparently it does!